src/main/java/io/cdap/plugin/aggregator/Pivot.java

```diff
@@ -0,0 +1,409 @@
+    if (!PivotConfig.ERROR_SKIP.equals(onError) && groupValue.getMissingColumnsModel().size() > 0) {
+      handleOnError(onError, groupValue.getMissingColumnsModel(), emitter, structuredRecord);
+      return;
+    }
+
+    StructuredRecord.Builder builder = StructuredRecord.builder(this.outputSchema);
+    List<Schema.Field> fields = this.outputSchema.getFields();
+
+    String pivotRow = config.getPivotRow();
+    String defaultValue = config.getDefaultValue();
+
+    Map<String, AggregateFunction> functions = groupValue.getFunctions();
+
+    for (Schema.Field field : fields) {
+      String fieldName = field.getName();
+      if (fieldName.equals(pivotRow)) {
+        builder.set(fieldName, structuredRecord.get(pivotRow));
+      } else {
+        String functionKey = findAggregationFunctionsKey(fieldName);
+        AggregateFunction aggregateFunction = functions.get(functionKey);
+        Object aggregateValue = aggregateFunction.getAggregate();
+        if (aggregateValue == null) {
+          if (defaultValue != null) {
+            builder.convertAndSet(fieldName, config.getDefaultValue());
+          }
+        } else {
+          builder.set(fieldName, aggregateValue);
+        }
+      }
+    }
+    emitter.emit(builder.build());
+  }
+
+  private void handleOnError(String onError, HashMap<String, Set<Object>> missingColumnsModel,
+                             Emitter<StructuredRecord> emitter, StructuredRecord inputRecord) throws Exception {
+    String errorMessage = missingColumnsModel.entrySet().stream()
+      .map(missingColumnValue -> {
+             String missingModels = missingColumnValue.getValue().
+               stream()
+               .map(String::valueOf)
+               .collect(Collectors.joining(" ,"));
+             return String.format("For columns name: %s following models are missing %s",
+                                  missingColumnValue.getKey(), missingModels);
+           }
+      ).collect(Collectors.joining(" ;"));
+
+    if (PivotConfig.ERROR_FAIL_PIPELINE.equals(onError)) {
+      emitter.emitAlert(ImmutableMap.of("Error", errorMessage));
+      throw new Exception(String.format("Failing pipeline due to error : %s", errorMessage));
+    }
+
+    if (PivotConfig.ERROR_SEND_TO_ERROR_PORT.equals(onError)) {
+      emitter.emitError(new InvalidEntry<>(0, errorMessage, inputRecord));
+    }
+  }
+
+  private void init(FailureCollector failureCollector) {
+    this.failureCollector = failureCollector;
+    pivotColumnsAndData = config.getPivotColumnsAndData(failureCollector);
+    orderedColumnsName = new ArrayList<>(pivotColumnsAndData.keySet());
+    functionInfos = config.getAggregates(failureCollector);
+    fieldAliases = config.getFieldAliases(failureCollector);
+    columnsProduct = generateColumnsProduct(pivotColumnsAndData);
+    failureCollector.getOrThrowException();
+  }
+
+  private void validate(Schema inputSchema, FailureCollector failureCollector) {
```

Comment:
why dont we move all this validation in config.validation method?

Reply:
DONE